### ISSUE TITLE
Add exporter for Citation File Format (CFF) (yaml based)

### ISSUE THREAD

**Issue by Siedlerchr** [JabRef/jabref#10661]
**Is your suggestion for improvement related to a problem? Please describe.**
JabRef should be able to export citations into the cff file format. There is already an implemented importer which maps bibtex fields.
The existing YAML exporter can be easily adapted.

**Additional context**

https://citation-file-format.github.io/

An example python converter
https://github.com/citation-file-format/cffconvert

**Comment by HarikaB11** [comment]
Hi @Siedlerchr, I would like to work on this issue.

**Comment by koppor** [comment]
Here are some hints_

TLDR: JabRef has the functionality to import CFF (https://github.com/JabRef/jabref/issues/7945), but no functionality for export. This should be implemented.

JabRef should be able to parse Citation File Format (CFF) and export into it.

Links:

- General CFF information: https://citation-file-format.github.io/ 
- Mapping BibTeX and CFF: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files#citing-something-other-than-software
- Python implementation of a converter: https://github.com/citation-file-format/cff-converter-python 
- Description of the BibTeX format: https://docs.jabref.org/advanced/fields

Implementation hints:

The export can either be done by Java code directly or creating `.layout` files.

## Export using `.layout` files:

See `yaml.*` in https://github.com/JabRef/jabref/tree/main/src/main/resources/resource/layout

Example: https://github.com/JabRef/jabref/blob/main/src/main/resources/resource/layout/yaml.layout

Documentation on .layout files: https://docs.jabref.org/collaborative-work/export/customexports

---

Acceptance criteria:

- At least `@article`  and `@inproceedings` need to be supported.

---

Note: If you dislike the .layout format, one can work on replacing that part of JabRef by Java templating engines. See https://github.com/koppor/jabref/issues/392 for details.

**Comment by ShailikaS** [comment]
Hi @Siedlerchr, I would like to work on this if this issue is still open?

**Comment by ThiloteE** [comment]
There hasn't been activity here since a month, but Harika has had commits in other repositories, so I assume this one here was abandoned. Hence, I will assign @ShailikaS.

**Comment by Siedlerchr** [comment]
No activity since a while

**Comment by garymejia** [comment]
Hello, could I be assigned this issue?

**Comment by Siedlerchr** [comment]
@garymejia  Are you still working on this? Or can other work on this

**Comment by JEANPRBT** [comment]
I can work on this one !

**Comment by ThiloteE** [comment]
@garymejia are you still on it?

**Comment by garymejia** [comment]
Hey @ThiloteE, I started working on this issue a bit late. I'm okay with being unassigned this issue since I was still figuring out the formatting requirements for cff. Sorry for the holdup!

**Comment by ThiloteE** [comment]
ok, then I will assign @JEANPRBT
